samples/notebooks/image-captioning-gcp/Image Captioning TF 2.0.ipynb

```diff
@@ -0,0 +1,893 @@
+    "    base_image=BASE_IMAGE,\n",
+    "    dependency=[kfp.compiler.VersionedDependency(name='matplotlib', version='3.1.0')],\n",
+    "    target_image=TARGET_IMAGE)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h3> Create and run pipeline </h3>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h4> Create pipeline </h4>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.pipeline(\n",
+    "    name='Image Captioning Pipeline',\n",
+    "    description='A pipeline that trains a model to caption images'\n",
+    ")\n",
+    "def caption_pipeline(\n",
+    "dataset_path=GCS_DATASET_PATH,\n",
+    "num_examples=30000,\n",
+    "epochs=20,\n",
+    "training_batch_size=64,\n",
+    "hidden_state_size=512,\n",
+    "vocab_size=5000,\n",
+    "embedding_dim=256,\n",
+    "preprocessing_batch_size=16,\n",
+    "preprocessing_output_dir='default',\n",
+    "tokenizing_output_dir='default',\n",
+    "training_output_dir='default',\n",
+    "validation_output_dir='default',\n",
+    "): \n",
+    "    \n",
+    "    preprocessing_img_task = preprocessing_img_op(\n",
+    "        dataset_path, \n",
+    "        output_dir=preprocessing_output_dir,\n",
+    "        batch_size=preprocessing_batch_size, \n",
+    "        num_examples=num_examples).apply(\n",
+    "        use_gcp_secret('user-gcp-sa'))\n",
+    "    \n",
+    "    tokenize_captions_task = tokenize_captions_op(\n",
+    "        dataset_path, \n",
+    "        preprocessing_img_task.output, \n",
+    "        output_dir=tokenizing_output_dir, \n",
+    "        top_k=vocab_size).apply(\n",
+    "        use_gcp_secret('user-gcp-sa'))\n",
+    "    \n",
+    "    model_train_task = model_train_op(\n",
+    "        dataset_path, \n",
+    "        preprocessing_img_task.output,\n",
+    "        tokenize_captions_task.output,\n",
+    "        train_output_dir=training_output_dir, \n",
+    "        valid_output_dir=validation_output_dir,\n",
+    "        batch_size=training_batch_size, \n",
+    "        embedding_dim=embedding_dim, \n",
+    "        units=hidden_state_size, \n",
+    "        epochs=epochs).apply(\n",
+    "        use_gcp_secret('user-gcp-sa'))\n",
+    "    \n",
+    "    predict_task = predict_op(\n",
+    "        dataset_path,\n",
+    "        tokenize_captions_task.output, \n",
+    "        model_train_task.output,\n",
+    "        preprocess_output_dir=preprocessing_output_dir,\n",
+    "        valid_output_dir=validation_output_dir,\n",
+    "        embedding_dim=embedding_dim,\n",
+    "        units=hidden_state_size).apply(\n",
+    "        use_gcp_secret('user-gcp-sa'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline_func = caption_pipeline"
```

Comment:
Is this cell necessary? maybe simply using 'caption_pipeline' in the following cell?

Reply:
Removed